client/src/test/scala/drt/client/components/StaffingUtilSpec.scala

```diff
@@ -0,0 +1,108 @@
+package drt.client.components
+
+import drt.client.services.JSDateConversions.SDate
+import utest._
+
+object StaffingUtilSpec extends TestSuite {
+
+  def tests = Tests {
+    "When StaffingUtilSpec" - {
+      "Given consecutiveDayForWeek" - {
+        "Then handle the date 2024-10-30 correctly" - {
+          val viewingDate = SDate("2024-10-30T00:00:00Z")
+          val result = StaffingUtil.consecutiveDayForWeek(viewingDate)
+
+          assert(result == Seq(
+            (SDate("2024-10-28T00:00:00Z"), "Monday"),
+            (SDate("2024-10-29T00:00:00Z"), "Tuesday"),
+            (SDate("2024-10-30T00:00:00Z"), "Wednesday"),
+            (SDate("2024-10-31T00:00:00Z"), "Thursday"),
+            (SDate("2024-11-01T00:00:00Z"), "Friday"),
+            (SDate("2024-11-02T00:00:00Z"), "Saturday"),
+            (SDate("2024-11-03T00:00:00Z"), "Sunday")
+          ))
+        }
+      }
+
+      "Given consecutiveDay" - {
+        "Then handle the date 2024-10-30 correctly" - {
+          val startDay = SDate("2024-10-30T00:00:00Z")
+          val result = StaffingUtil.consecutiveDay(startDay)
+
+          assert(result == Seq(
+            (SDate("2024-10-30T00:00:00Z"), "Wednesday")
+          ))
+        }
+      }
+    }
+    "Given navigationDates" - {
+      "When isWeekly is true" - {
+        "Then handle the date 2024-10-30 correctly" - {
+          val viewingDate = SDate("2024-10-30T00:00:00Z")
+          val isWeekly = true
+          val isDaily = false
+
+          val (previousWeekDate, nextWeekDate) = StaffingUtil.navigationDates(viewingDate, isWeekly, isDaily, () => SDate("2024-10-30T00:00:00Z"))
+          assert(previousWeekDate.millisSinceEpoch == SDate("2024-10-22T23:00:00.00Z").millisSinceEpoch)
+          assert(nextWeekDate.millisSinceEpoch == SDate("2024-11-06T00:00:00Z").millisSinceEpoch)
+        }
+
+        "Then handle the date 2024-03-29 correctly when view date is in 6th month ahead" - {
```

Comment:
Maybe these test cases should be UTC & BST dates as that's the only difference I can think of that might affect the behaviour, rather than saying 6 months ahead